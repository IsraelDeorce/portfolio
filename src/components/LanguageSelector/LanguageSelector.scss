.flag {
  height: 24px;
  width: 48px;
}
