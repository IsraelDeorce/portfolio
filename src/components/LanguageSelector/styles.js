import styled from 'styled-components'

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
  color: ${({ theme }) => theme.c_brackets};
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`