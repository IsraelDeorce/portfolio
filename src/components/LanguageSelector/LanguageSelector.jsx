import { useState, useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { MdKeyboardArrowDown } from 'react-icons/md'
import Flags from 'country-flag-icons/react/3x2'
import { formatLanguage } from '../../utils'
import * as S from './styles'
import './LanguageSelector.scss'

function LanguageSelector() {
  const { language } = useSelector((state) => state.session)
  const dispatch = useDispatch()
  const [flag, setFlag] = useState(null)
  const [languageCode, setLanguage] = useState('')

  useEffect(() => {
    const formatedLanguage = formatLanguage(language)
    setLanguage(formatedLanguage)
    formatedLanguage === 'en'
      ? setFlag(<Flags.CA title='Canada' className='flag' />)
      : setFlag(<Flags.BR title='Brasil' className='flag' />)
  }, [language])

  const handleChangeLanguage = () => {
    dispatch.session.changeLanguage({
      language: languageCode === 'en' ? 'pt' : 'en'
    })
  }

  return (
    <S.LanguageSelector onClick={handleChangeLanguage}>
      <S.LanguageCode>
        {languageCode}
        {<MdKeyboardArrowDown />}
      </S.LanguageCode>
      {flag}
    </S.LanguageSelector>
  )
}

export default LanguageSelector
