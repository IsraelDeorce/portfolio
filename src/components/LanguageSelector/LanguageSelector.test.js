import LanguageSelector from './LanguageSelector'

describe('<LanguageSelector/>', () => {
  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<LanguageSelector />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<LanguageSelector />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<LanguageSelector />)).toMatchSnapshot()
    })
  })
})
