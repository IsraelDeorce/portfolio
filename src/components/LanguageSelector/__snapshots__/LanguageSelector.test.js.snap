// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelector/> Snapshot renders without crashing using mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<LanguageSelector>
  <styled.div
    onClick={[Function]}
  >
    <div
      className="c0"
      onClick={[Function]}
    >
      <styled.div>
        <div
          className="c1"
        >
          en
          <MdKeyboardArrowDown>
            <IconBase
              attr={
                Object {
                  "viewBox": "0 0 24 24",
                }
              }
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                  key="0"
                />
              </svg>
            </IconBase>
          </MdKeyboardArrowDown>
        </div>
      </styled.div>
      <CA
        className="flag"
        title="Canada"
      >
        <svg
          className="flag"
          viewBox="0 0 512 341.3"
        >
          <title>
            Canada
          </title>
          <path
            d="M0 0h512v341.3H0z"
            fill="#FFF"
          />
          <g
            fill="#D80027"
          >
            <path
              d="M0 0h142v341.3H0zM370 0h142v341.3H370zM306.5 206l50.4-25.2-25.2-12.6V143l-50.4 25.2 25.2-50.4h-25.2L256.1 80l-25.2 37.8h-25.2l25.2 50.4-50.4-25.2v25.2l-25.2 12.6 50.4 25.2-12.6 25.2h50.4V269h25.2v-37.8h50.4z"
            />
          </g>
        </svg>
      </CA>
    </div>
  </styled.div>
</LanguageSelector>
`;

exports[`<LanguageSelector/> Snapshot renders without crashing using render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
      />
    </svg>
  </div>
</div>
`;

exports[`<LanguageSelector/> Snapshot renders without crashing using shallow 1`] = `
<styled.div
  onClick={[Function]}
>
  <styled.div>
    <MdKeyboardArrowDown />
  </styled.div>
</styled.div>
`;
